### DISCUSSION TITLE
can poetry delete virtual enviroments created outside poetry?

### DISCUSSION THREAD

**Discussion by sandeep026** [python-poetry/poetry#9483] [Q&A] [answered]
I have the following virtual environments, where  venv was created without poetry. 

```
.venv (Activated)
calc-u2IeJV_l-py3.11
```

I get the following errors when I try to delete it.

```
poetry env remove .venv
```

```
[WinError 2] The system cannot find the file specified
```

```
[WinError 5] Access is denied
```

Can environments created without poetry be deleted using poetry?

**Comment by radoering** [comment, marked as answer]
At least, it is _not_ intended to work for environments not created by Poetry. See https://python-poetry.org/docs/managing-environments/#deleting-the-environments valid examples for `env remove`.

However, I think `.venv` is a special case because it can be created by Poetry (see https://python-poetry.org/docs/configuration/#virtualenvsin-project). Maybe, passing the path to the Python executable or running `poetry env remove --all` works in that case. Maybe, you have to set `virtualenvs.in-project` first.
